File: output/CodeLlama/codenet/C++/Python/s522742913.py
import math
import sys

source = sys.stdin.readline()

if len(source) > 10 or len(source) < 1:
    sys.exit(-1)

max_i = int(math.pow(2, len(source) - 1))

sum = 0

for i in range(max_i):
    plus = bin(i)[2:]
    plus = plus.zfill(len(source))

    number = 0

    working = source
    while len(plus) > 0:
        i = 0
        while plus[i] == '0':
            i += 1
        sum += float(working[0:i + 1])
        plus = plus[i + 1:]
        working = working[i + 1:]
    sum += float(working)

print('%.0f' % sum)

File: output/CodeLlama/codenet/C++/Java/s522742913.java
import java.util.Scanner;

public class s522742913 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String source = sc.nextLine();
        if (source.length() > 10 || source.length() < 1) {
            return;
        }
        int max_i = (int) Math.pow(2, source.length() - 1);
        double sum = 0;
        for (int i = 0; i < max_i; i++) {
            String plus = Integer.toBinaryString(i);
            while (plus.length() < source.length()) {
                plus = "0" + plus;
            }
            double number = 0;
            String working = source;
            for (int j = 0; j < plus.length(); j++) {
                if (plus.charAt(j) == '1') {
                    number += Double.parseDouble(working.substring(0, j + 1));
                    working = working.substring(j + 1);
                }
            }
            number += Double.parseDouble(working);
            sum += number;
        }
        System.out.printf("%.0f", sum);
    }
}